src/main/java/com/commercetools/sync/products/ProductSync.java

```diff
@@ -186,11 +187,16 @@ private void processFetchedProducts(@Nonnull final Set<Product> fetchedProducts)
     }
 
     @Nonnull
-    private CompletionStage<List<Optional<Product>>> createOrUpdateProducts() {
-        return productService.createProducts(draftsToCreate)
-                             .thenAccept(createdProducts ->
-                                 updateStatistics(createdProducts, draftsToCreate.size()))
-                             .thenCompose(ignoredResult -> syncProducts(productsToSync));
+    private CompletableFuture<Void> createAndUpdateProducts() {
+        final CompletableFuture<Void> createRequestsStage =
+            productService.createProducts(draftsToCreate)
+                          .thenAccept(createdProducts -> updateStatistics(createdProducts, draftsToCreate.size()))
+                          .toCompletableFuture();
+
+        final CompletableFuture<List<Optional<Product>>> updateRequestsStage =
+            syncProducts(productsToSync).toCompletableFuture();
+
+        return allOf(createRequestsStage, updateRequestsStage);
```

Comment:
Did you notice any performance difference between previous and current approach as i doubt there will be any?

Reply:
Actually I haven't noticed yet. Maybe I will notice on the benchmarks. But why do you doubt there will be any difference? Because of the number of parallel requests set on the client?